server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java

```diff
@@ -198,6 +198,7 @@ public final class IndexScopedSettings extends AbstractScopedSettings {
                 IndexSettings.INDEX_MERGE_ON_FLUSH_MAX_FULL_FLUSH_MERGE_WAIT_TIME,
                 IndexSettings.INDEX_MERGE_ON_FLUSH_POLICY,
                 IndexSettings.DEFAULT_SEARCH_PIPELINE,
+                IndexSettings.INDEX_CONCURRENT_SEGMENT_SEARCH_SETTING,
```

Comment:
Lets use the [FeatureFlagged setting mechanism](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/common/settings/IndexScopedSettings.java#L230) to register these new settings only when FeatureFlag is enabled